### PR TITLE
React test library heba

### PR DESCRIPTION
**Pull request by hebasaadosman**
many cases of test :

-mock axios 
-mock fetch
-mock functions
-mock component
-mock more than one instance of component
-mock routes 
-mock component with children
-add RTL for atoms(avatar - buttons - checkbox - inputs - radio - toggle
this pr includes all files and cases i  have  tested so change in files is so  large and some files from pr #30 to able to test them .
 I also  added  some packages to help me in testing :

- "axios": "^0.21.1",  jest-mock-axios": "^4.2.1"  to test axios and mock it.
 
-  "react-router": "^5.2.0", "react-router-dom": "^5.2.0",  to test route.

### CODE REVIEW

File: storybook/package.json [hunk 1]
```diff
@@ -31,11 +35,6 @@
     "*.{jsx,js}": "eslint --cache --fix",
     "*.{js,css,md}": "prettier --write"
   },
-  "husky": {
```
Comment (Safei-Ashraf) [inline storybook/package.json:40]: please do NOT remove the pre-commit hooks.

File: storybook/src/RTL-heba/Mock/Mock.react.router.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React, { Component } from "react";
```
Comment (Safei-Ashraf) [inline storybook/src/RTL-heba/Mock/Mock.react.router.test.js:1]: image
test fails.

**Review by Safei-Ashraf — CHANGES_REQUESTED** [review]
You have mentioned changes to 7 files in the PR desc section.
it seems like you have around 57 files changes, including stories being added and adjustment being made to some react comps.
Please include details of the changes you made and make sure they match the changes you have already done. the PR includes much more than the test cases you have mentioned above so I can carry on with the review. you can view those at the "files changes" tab on the top of your branch
Also, make sure to re-enable the commit hooks before committing your code for the 2nd time.

File: storybook/src/RTL-heba/Mock/Mock.react.router.test.js (hunk 2 above)
Comment (hebasaadosman) [reply storybook/src/RTL-heba/Mock/Mock.react.router.test.js:1]: react-route in package.json try to "yarn"

File: storybook/package.json (hunk 1 above)
Comment (hebasaadosman) [reply storybook/package.json:40]: ok

**Comment by hebasaadosman** [comment]
safi,, thanks for your time and your professional notes  
 1-I edit the pr  description .
 2- I think if you give yourself chance to view my code and the size of test , you will understand the size and  importance of all these changes

**Comment by Safei-Ashraf** [comment]
1-PR description has to match the changes made, or the PR can NOT be reviewed properly.
2- 57 files changed on this branch, 31 of them has nothing to do with tests, they are either components or stories, if you added extra components to test, you have to include it into the PR descriptions.
3-Changes made to config files, pkg.json, new pkgs installed need to be mentioned in your PR summary.
Appreciate the time you put into your work, yet it has descriptive as much as possible.

**Review by Safei-Ashraf — APPROVED** [review]

**Comment by Safei-Ashraf** [comment]
LGTM

**Pull request merged**